### PR TITLE
Document the supported metrics for telemetry

### PR DESCRIPTION
**Pull request by mchurichi**
This PR adds a document that list all of the metrics that are being emitted by SPIRE.

Signed-off-by: Maximiliano Churichi <maximiliano.churichi@hpe.com>

**Pull Request check list**

- [x] Commit conforms to CONTRIBUTING.md?
- [ ] Proper tests/regressions included?
- [x] Documentation updated?

**Affected functionality**

No affected functionality

**Description of change**

Improves documentation

**Which issue this PR fixes**

No issue

### CODE REVIEW

File: doc/telemetry.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
+# Telemetry
+
+The SPIRE Server and Agent can be configured to emit metrics that can be send to the supported metrics collectors. For instructions on how to configure them properly please refer to the [Telemetry Configuration](telemetry_config) guide.
```
Comment (amartinezfayo) [inline doc/telemetry.md:3]: The .md extension is missing in the link.

```suggestion
The SPIRE Server and Agent can be configured to emit metrics that can be sent to the supported metrics collectors. For instructions on how to configure them properly, please refer to the [Telemetry Configuration](telemetry_config.md) guide.
```

File: doc/telemetry.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+The following metrics are emited:
```
Comment (amartinezfayo) [inline doc/telemetry.md:5]:
```suggestion
The following metrics are emitted:
```

File: doc/telemetry.md (hunk 1 above)
Comment (mchurichi) [reply doc/telemetry.md:3]: good catch!

**Comment by dfeldman** [comment]
When ingested into statsd, the metrics look like this: 
"spire_agent_agent_svid_rotate_Internal_host-agent-0" 
(Where host-agent-0 is the hostname)
Maybe we can document this format somewhere?

**Comment by mchurichi** [comment]
> When ingested into statsd, the metrics look like this:
> "spire_agent_agent_svid_rotate_Internal_host-agent-0"
> (Where host-agent-0 is the hostname, spire-agent is the service name, and "internal" is .... something :) )
> Maybe we can document this format somewhere?

@dfeldman Sure, sgtm, do you know where I can find an explanation of the format?

**Comment by amartinezfayo** [comment]
Talked offline with @mchurichi that it would be good to have descriptions of the metrics. Also, the `node_api` and `registration_api` are part of the old APIs but we can keep them as we support them.
This documentation will need to be updated with the metrics emitted by the refactored APIs.

**Review by amartinezfayo — COMMENTED** [review]
Thank you @mchurichi, it looks good!
I have only a few small nits.

**Review by amartinezfayo — APPROVED** [review]
Thank you for this @mchurichi!

**Pull request merged**